nntrainer/src/bn_layer.cpp

```diff
@@ -37,10 +38,10 @@ int BatchNormalizationLayer::initialize(bool last) {
   dim = input_dim;
   output_dim = dim;
 
-  this->gamma = Tensor(dim.channel(), dim.batch(), dim.width());
-  this->beta = Tensor(dim.channel(), dim.batch(), dim.width());
-  beta.setZero();
-  gamma.setZero();
+  this->mu = Tensor(dim.channel(), dim.height(), dim.width());
+  this->var = Tensor(dim.channel(), dim.height(), dim.width());
```

Comment:
Do we need to store these variables as class variables? They are only used locally for forwarding.
The same goes for saving them as well.

IMO, we store the moving average of these variables which are used in the `EVAL` mode, not these variables directly.

Reply:
mu and var needs to be updated and saved through training and this mu and var is used when inference.
More patches needed for that. (eg. adding momentum properties, save & read momentum to the file, using saved mu / var when infer... )

Changing `mu` to `moving_mu`  could be done though, IMO it is unnecessary since only `moving_mu` is used class wide(and for `var`)

Reply:
aah, you intend `mu` to store the moving average. 

> Changing mu to moving_mu could be done though, IMO it is unnecessary since only moving_mu is used class wide(and for var)

Yeah, how about just add some comment at its declaration about this?

Reply:
Can do. Thanks for the suggestion :)